frame/dapps-staking/src/tests.rs

```diff
@@ -1205,3 +1209,418 @@ fn new_era_forcing() {
         )));
     })
 }
+
+#[test]
+fn claim_contract_not_registered() {
+    ExternalityBuilder::build().execute_with(|| {
+        let claimer = 2;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+
+        assert_noop!(
+            DappsStaking::claim(Origin::signed(claimer), contract),
+            crate::pallet::pallet::Error::<TestRuntime>::ContractNotRegistered
+        );
+    })
+}
+
+#[test]
+fn claim_nothing_to_claim() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let claimer = 2;
+        const ERA_REWARD: Balance = 1000;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, 0);
+        let start_era = DappsStaking::current_era().unwrap_or(Zero::zero());
+        register(developer1, contract.clone());
+
+        assert_noop!(
+            DappsStaking::claim(Origin::signed(claimer), contract),
+            crate::pallet::pallet::Error::<TestRuntime>::NothingToClaim
+        );
+    })
+}
+
+#[test]
+fn claim_twice_in_same_era() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let claimer = 2;
+        const ERA_REWARD: Balance = 1000;
+        const STAKE_AMOUNT: Balance = 100;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+        const SKIP_ERA: EraIndex = 3;
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, 0);
+        let start_era = DappsStaking::current_era().unwrap_or(Zero::zero());
+        register(developer1, contract.clone());
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, 0);
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(claimer, contract, start_era, claim_era.clone());
+
+        assert_noop!(
+            DappsStaking::claim(Origin::signed(claimer), contract),
+            crate::pallet::pallet::Error::<TestRuntime>::AlreadyClaimedInThisEra
+        );
+    })
+}
+
+#[test]
+fn claim_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let claimer = 2;
+        const ERA_REWARD: Balance = 1000;
+        const STAKE_AMOUNT: Balance = 100;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+        const SKIP_ERA: EraIndex = 3;
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, 0);
```

Comment:
It is misleading that FROM_ERA implies that you are skipping FROM_ERA eras. It doesn't make sense, imo.